api/posts/posts-router.js

```diff
@@ -1 +1,112 @@
 // implement your posts router here
+const express = require(`express`)
+const Posts = require(`./posts-model`)
+const router = express.Router()
+
+//GET Requests:
+router.get(`/`, (req, res) => {
+   Posts.find()
+      .then(posts => {
+         res.status(200).json(posts)
+      })
+      .catch(err => {
+         console.log(err)
+         res.status(500).json({ message: "The posts information could not be retrieved" })
+      })
+})
+
+router.get(`/:id`, (req, res) => {
+   const { id } = req.params
+
+   Posts.findById(id)
+      .then(post => {
+         if (!post) {
+            res.status(404).json({ message: "The post with the specified ID does not exist" })
+         }
+         else (res.status(200).json(post))
+
+      })
+      .catch(err => {
+         console.log(err)
+         res.status(500).json({ message: "The post information could not be retrieved" })
+      })
+})
+
+router.get(`/:id/comments`, (req, res) => {
+   Posts.findPostComments(req.params.id)
+      .then(postComments => {
+         if (postComments.length === 0) {
+            return res.status(404).json({ message: "The post with the specified ID does not exist" })
+         }
+         res.status(200).json(postComments)
+      })
+      .catch(err => {
+         console.log(`Error:`, err)
+
+         res.status(500).json({ message: "The comments information could not be retrieved" })
+      })
+})
+
+//POST Request:
+router.post(`/`, (req, res) => {
+   Posts.insert(req.body)
```

Comment:
Nit: It would be best to check the req.body for the title and contents and through the error before even trying to create a new post